rasa/core/actions/flows.py

```diff
@@ -164,3 +165,51 @@ async def run(
         events.extend([SlotSet(k, v) for k, v in corrected_slots.items()])
 
         return events
+
+
+class ActionClarifyFlows(action.Action):
+    """Action which clarifies which flow to start."""
+
+    def name(self) -> Text:
+        """Return the flow name."""
+        return ACTION_CLARIFY_FLOWS
+
+    @staticmethod
+    def assemble_options_string(names: List[str]) -> str:
+        """Concatenate options to a human-readable string."""
+        clarification_message = ""
+        for i, name in enumerate(names):
+            if i == 0:
+                clarification_message += name
+            elif i == len(names) - 1:
+                clarification_message += f" or {name}"
+            else:
+                clarification_message += f", {name}"
+        return clarification_message
+
+    async def run(
+        self,
+        output_channel: "OutputChannel",
+        nlg: "NaturalLanguageGenerator",
+        tracker: "DialogueStateTracker",
+        domain: "Domain",
+        metadata: Optional[Dict[Text, Any]] = None,
+    ) -> List[Event]:
+        """Correct the slots."""
+        stack = FlowStack.from_tracker(tracker)
+        if stack.is_empty():
+            structlogger.warning("action.clarify_flows.no_active_flow", stack=stack)
+            return []
+
+        top_frame = stack.top()
+        if top_frame is None or top_frame.flow_id != "pattern_clarification":
+            structlogger.warning(
+                "action.clarify_flows.clarify_frame_not_on_top", stack=stack
+            )
+            return []
+
+        context = top_frame.context
+        names = context.get("names", [])
+        options_string = self.assemble_options_string(names)
+        context["clarification_options"] = options_string
```

Comment:
uuuh, i think that is a no go 😂 no guarantee that will work

Reply:
I mean it's like a slot set using the context, and it does work (at least as of now). Also I don't know how else I would do it without introducing a slot for it, which I could do. I just tried it this way since we currently don't really have slots for the default patterns. Maybe something we need to explore a bit / discuss.

Reply:
Aaaah you are storing the stack in the slot again, didn’t realize that - I thought you were just modifying the dict